### PR TITLE
U805-023: use previous context in signatureHelp

### PR DESCRIPTION
**Pull request by AdrienBoulanger**
If available try to reuse the signatureHelpContext which contains
the previous signatures: it's faster to just filter them that
recompute a new list of signatures.

Edge case for "\b": removing a character can make more signatures
valid and they could not be contained in the previous signature =>
always recompute the list.

Adapt tests: retriggerCharacters have changed
Add a new test.

### CODE REVIEW

File: source/ada/lsp-ada_handlers.adb [hunk 1]
```diff
@@ -2494,11 +2494,33 @@ package body LSP.Ada_Handlers is
...
       procedure Add_Signature (Decl_Node : Libadalang.Analysis.Basic_Decl);
+      --  Add the signature of Decl_Node if it matches the current context.
+
+      procedure Filter_Signature
+        (Signature_Info : SignatureInformation;
+         Designator     : Libadalang.Analysis.Ada_Node;
+         Position       : LSP.Types.LSP_Number;
+         Is_Active      : Boolean);
+      --  Filter the signatures returned by the client using the current
+      --  current Position and designator.
```
Comment (AnthonyLeonardoGracio) [inline source/ada/lsp-ada_handlers.adb:2511]: should be removed

**Review by AnthonyLeonardoGracio — CHANGES_REQUESTED** [review]

File: source/ada/lsp-ada_handlers.adb (hunk 1 above)
Comment (AdrienBoulanger) [reply source/ada/lsp-ada_handlers.adb:2511]: Remove the comment or replace "filter" by "remove"
